### PR TITLE
Fix/do not send non metric measurements

### PR DESCRIPTION
**Pull request by corycaywood**
# Description 
The original implementation of preventing measurements from being added to the buffer when there isn't a metric seems to have some thread issues. Sometimes measurements get added even when there is no longer a metric. This fix prevents any measurements from being sent by the `Sender` with a null `Metric`.

## Links
APTT-291

# Checklist
- [x] I have read the [contributing guidelines](../CONTRIBUTING.md).
- [x] I wrote/updated tests for new/changed code
- [x] I ran `./gradlew check` without errors

### CODE REVIEW

File: Core/src/test/java/com/rakuten/tech/mobile/perf/core/SenderSpec.java
Comment (NemoOudeis) [inline Core/src/test/java/com/rakuten/tech/mobile/perf/core/SenderSpec.java]: I'm a bit confused by the test setup, it reads like:
* sender with only sending measurements attached to metrics
* setup 10 metrics in the buffer
* setup 20 custom measurements

now the confusing part to me is: why 10 metrics? wouldn't 1 metric me more intuitive?

File: Core/src/test/java/com/rakuten/tech/mobile/perf/core/SenderSpec.java
Comment (NemoOudeis) [inline Core/src/test/java/com/rakuten/tech/mobile/perf/core/SenderSpec.java]: all the `protected`s could be package private (claims my IDE) which would make the code slightly more readable

**Review by NemoOudeis — CHANGES_REQUESTED** [review]
2 small issues, i'd prefer you'd change those before we merge.
functionally it lgtm

File: Core/src/test/java/com/rakuten/tech/mobile/perf/core/SenderSpec.java
Comment (corycaywood) [reply Core/src/test/java/com/rakuten/tech/mobile/perf/core/SenderSpec.java]: Yeah, you're right. I copied what the other tests were doing, but didn't pay close enough attention.

File: Core/src/test/java/com/rakuten/tech/mobile/perf/core/SenderSpec.java
Comment (corycaywood) [reply Core/src/test/java/com/rakuten/tech/mobile/perf/core/SenderSpec.java]: Ah, okay I will update this.

**Review by NemoOudeis — APPROVED** [review]

**Comment by NemoOudeis** [comment]
good to merge after rebase onto master

**Pull request merged**
